.github/workflows/deploy.yml

```diff
@@ -1,49 +1,52 @@
-# This is a basic workflow to help you get started with Actions
-
 name: CI
 
-# Controls when the workflow will run
 on:
-  # Triggers the workflow on push or pull request events but only for the dev branch
   push:
     branches: [ dev ]
   pull_request:
     branches: [ dev ]
 
-  # Allows you to run this workflow manually from the Actions tab
   workflow_dispatch:
 
-# A workflow run is made up of one or more jobs that can run sequentially or in parallel
 jobs:
-  # This workflow contains a single job called "build"
   build:
-    # The type of runner that the job will run on
     runs-on: ubuntu-latest
-
-    # Steps represent a sequence of tasks that will be executed as part of the job
-    # .github/workflows/app.yaml
     steps:
-      - name: Check out repository code
-        uses: actions/checkout@v2
-
-      # Setup Python (faster than using Python container)
-      - name: Setup Python
-        uses: actions/setup-python@v2
-        with:
-          python-version: "3.x"
-
-      - name: download poetry
+      - uses: actions/checkout@v2
+      - name: Create File app/.env
         run: |
-          curl https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py > get-poetry
-      - id: Install_poetry
-        run: python get-poetry
-
-      - name: activate poetry
+          echo SECRET_KEY='146EtHWvxaInR3YB3d2Cdtn4emQcUBWw' > app/.env
+          echo FLASK_APP='main' >> app/.env
+          echo FLASK_ENV='development' >> app/.env
+
+          echo DATABASE='postgresql' >> app/.env
+          echo SQL_HOST='db' >> app/.env
+          echo SQL_USER='flask_user' >> app/.env
+          echo SQL_PASSWORD='1234567816' >> app/.env
+          echo SQL_DB_NAME='video_streaming' >> app/.env
+          echo SQL_PORT='5432' >> app/.env
+      - name: Show file app/env content
+        run: |
+          pwd
+          ls -la
+          cat app/.env
+      - name: Create File app/env.db
         run: |
-          export PATH = "${PATH}:/root/.poetry/bin"
-      - name: install dependencies
+          echo POSTGRES_USER=flask_user > app/.env.db
+          echo POSTGRES_PASSWORD=1234567816 >> app/.env.db
+          echo POSTGRES_DB=video_streaming >> app/.env.db
+      - name: Show file app/env.db content
         run: |
-          poetry config virtualenvs.create false
-          poetry install --no-dev
-#      - name: run tests
-#        run: poetry run pytest
+          pwd
+          ls -la
+          cat app/.env.db
+      - name: Build the stack
+        run: docker-compose up -d
+      - name: Test
+        env:
+          COMPOSE_INTERACTIVE_NO_CLI: 1
+        run: docker-compose exec -T web pytest
+      - name: PreCommit install
```

Comment:
use separate GH actions for test and pre-commit

Reply:
done